### DISCUSSION TITLE
locode Code first approach

### DISCUSSION THREAD

**Discussion by tiptopweb** [ServiceStack/Discuss#17] [Q&A] [answered]
Hello
I am trying to use code first approach (my get 6.0.3)
My boolean or date types are not decoded by locode
I was under the impression it was automatic?

![types-not-decoded](https://user-images.githubusercontent.com/5791758/161489672-5ac11871-9482-4d5c-822d-4e6299624b75.png)

I tried to also add the Date format, just in case
https://github.com/tiptopweb/astro/blob/main/api/Tiptopweb.Astro/Configure.AutoQuery.cs

Here are my types
https://github.com/tiptopweb/astro/blob/main/api/Tiptopweb.Astro.ServiceModel/Types/Article.cs

All the code is visible in Github

**Comment by Layoric** [comment, marked as answer]
The Astro project is still using [`GeneratedServices` ](https://github.com/tiptopweb/astro/blob/main/api/Tiptopweb.Astro/Configure.AutoQuery.cs#L23), for it to be code first, you would have to declare your AutoQuery DTOs as well as remove the use of `GenerateCrudServices`.

```csharp
public class QueryImage : QueryDb<Image>
{
    
}

public class CreateImage : ICreateDb<Image>, IReturn<Image>
{
    public long ArticleId { get; set; }
    
    public string UrlProxy { get; set; }    
    
    public bool Published { get; set; }
    
    public bool IsFeatured { get; set; }
    
    public bool Deleted { get; set; }
    
    public long Status { get; set; }
    
    public long Type { get; set; }
    
    public long DisplayOrder { get; set; }
        
    public string Name { get; set; }  
}
```

And since the DTO has the required metadata of `bool`, Locode can infer the checkbox in the UI.

![image](https://user-images.githubusercontent.com/234642/161499470-eda80c1d-53a6-4226-bb13-bf2fce9fa384.png)

The reason generated services doesn't show as a checkbox is because SQLite doesn't have [a separate boolean](https://www.sqlite.org/datatype3.html) type, values are stored as integers which are presented as number fields in the UI.
